crates/ide_assists/src/handlers/add_explicit_type.rs

```diff
@@ -258,6 +258,25 @@ fn main() {
 fn main() {
     let test @ (): () = ();
 }
+"#,
+        );
+    }
+
+    #[test]
+    fn add_explicit_type_inserts_coercions() {
+        // This should set `*const [u32]`
+        check_assist(
+            add_explicit_type,
+            r#"
+//- minicore: coerce_unsized
+fn f() {
+    let $0x: *const [_] = &[3];
+}
+"#,
+            r#"
+fn f() {
+    let x: *const [_] = &[3];
+}
 "#,
         );
     }
```

Comment:
So for some reason the coercions here are all slices of type variables instead of a conrete type(even if using a `3u32` literal). Is this expected?

Reply:
Oh, we need to resolve the types in adjustments after inference, see this function: https://github.com/rust-analyzer/rust-analyzer/blob/6f3a3bdde4e526d880b9d8d9c613d2cef50a5f3b/crates/hir_ty/src/infer.rs#L390

Reply:
Oooh ye that totally makes sense. Though out of interest why do we lose the concrete type information for the array/slice there even if it is known before the coercion? I would expect that to persist in this specific case.

Reply:
The type we coerce *to* is `*const [_]`, which contains a type variable. The type variable gets unified with the `u32`, but to actually see that we have to resolve it in the end.